lib/puppet/pops/migration/migration_checker.rb

```diff
@@ -6,6 +6,10 @@ class Puppet::Pops::Migration::MigrationChecker
   def initialize()
   end
 
+  def self.singleton
+    @@null_checker ||= self.new
```

Comment:
This is a class variable on a class method. Isn't a class instance variable more appropriate?